client/src/container/HomePage/index.js

```diff
@@ -1,2 +1,147 @@
-const HomePage = () => {};
+import React, { useState, useEffect } from 'react';
+import propTypes from 'prop-types';
+import {
+  notification,
+  Button,
+  Spin,
+  Rate,
+  TreeSelect,
+  Input,
+  Empty,
+} from 'antd';
+import { AudioOutlined } from '@ant-design/icons';
+
+import axios from 'axios';
+import categories from '../../assets/categories';
+import './style.css';
+
+const { Search } = Input;
+const suffix = (
+  <AudioOutlined
+    style={{
+      fontSize: 16,
+      color: '#1890ff',
+    }}
+  />
+);
+
+const HomePage = ({ history }) => {
+  const [loading, setLoading] = useState(true);
+  const [topcourses, setTopCourses] = useState([]);
+  const [searchCourseName, setSearchCourseName] = useState('');
+  const [cat, setCat] = useState('');
+
+  const fetchCoursesByNameAndCatId = async (catId, courseName) => {
+    try {
+      const { data } = await axios.post(`/api/v1/catId/courseName`, {
+        catId,
+        courseName,
+      });
+      setTopCourses(data);
+    } catch (err) {
+      let error;
+      if (err.response) error = err.response.data.msg || 'Something went wrong';
+      else error = 'No Course with this name';
+      notification.error({ message: error });
+    }
+  };
+
+  const fetchTopCourses = async () => {
+    try {
+      const { data } = await axios.get(`/api/v1/topCourses`);
+      setTopCourses(data);
+      setLoading(false);
+    } catch (err) {
+      let error;
+      if (err.response) error = err.response.data.msg || 'Something went wrong';
+      else error = 'No Courses';
+      notification.error({ message: error });
+    }
+  };
+  const handleClick = (id) => {
+    history.push(`/course/${id}`);
+  };
+  const treeSelectOnChange = (value) => {
+    if (searchCourseName) {
+      setCat(value);
+      fetchCoursesByNameAndCatId(value, searchCourseName);
+    }
+  };
+  const inputOnSearch = (value) => {
+    if (value.value) {
+      setSearchCourseName(value);
+      fetchCoursesByNameAndCatId(cat, value);
+    }
+  };
+  const inputOnChange = (e) => {
+    setSearchCourseName(e.target.value);
+    fetchCoursesByNameAndCatId(cat, e.target.value);
+  };
+  useEffect(() => {
+    fetchTopCourses();
+  }, []);
+
+  return (
+    <div>
+      <div className="search-container">
+        <TreeSelect
+          style={{ width: '20%', marginRight: '10px' }}
+          // value={cat}
+          dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}
+          treeData={categories}
+          onChange={treeSelectOnChange}
+          placeholder="Please select"
+          treeDefaultExpandAll
+        />
+
+        <Search
+          style={{ width: '20%' }}
+          placeholder="input search text"
+          onSearch={inputOnSearch}
+          onChange={inputOnChange}
+          enterButton
+          suffix={suffix}
+        />
+      </div>
+
+      <div className="topRate__container">
+        {loading ? (
+          <Spin />
+        ) : topcourses.length > 0 ? (
```

Comment:
**you need to handle error case here :** 

`{error ? (use antd component for error) : loading ? (<Spin /> ) : topcourses.length ? () : () }`